src/batchUploader.ts

```diff
@@ -59,27 +79,55 @@ export class BatchUploader {
     }
 
     private isBeaconAvailable(): boolean {
+        // TODO: Is there any chance that `sendBeacon` exists
+        //       but that the beacon would not be available?
         if (navigator.sendBeacon) {
             return true;
         }
         return false;
     }
 
+    // Triggers a setTimeout for prepareAndUpload
+    private triggerUploadInterval(
+        triggerFuture: boolean = false,
+        useBeacon: boolean = false
+    ): void {
+        // Clear out existing interval before creating a new one
+        clearTimeout(this.uploadIntervalTimeout);
```

Comment:
I see what you are trying to do here, but `clearTimeout` accepts a timeout id, and not a number of seconds. You can have multiple timeouts, and each one returns a timeout id.  The individual timeouts would need to be set to different variables which would hold the timeout id and then you can pass the variable to `clearTimeout`.   [For reference](https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout).

If I am reading this right, it should be:
```
clearTimeout(this.uploadIntervalTimeout)
```